example/prometheus-operator-crd/monitoring.coreos.com_prometheuses.yaml

```diff
@@ -4486,6 +4486,128 @@ spec:
                 baseImage:
                   description: Thanos base image if other than default.
                   type: string
+                grpcServerTlsConfig:
+                  description: 'GRPCServerTLSConfig configures the GRPC server from
+                    which Thanos Querier reads recorded rule data. Note: currently
+                    only the CAFile, CertFile, and KeyFile fields are supported. Maps
```

Comment:
What else is not? :thinking:

Reply:
also I think we can remove this line. We even have `insecureSkipVerify`

Reply:
I'm re-using the TLSConfig type which is used in a few places to configure TLS for prometheus endpoints.  In addition to the file options, it also includes fields `CA`, `Cert`, and `KeySecret` which allow direct usage of secrets or config maps instead of needing to manually mount the secret.  So we can add support for this to the Thanos types at a later point if we want.

The type also includes insecureSkipVerify which is not available as a CLI arg to Thanos so it's ignore if it's set here.

Reply:
:+1: